src/util.rs

```diff
@@ -802,27 +802,36 @@ impl Run {
                 )
             }
             RunMode::Egglog => {
-                assert_eq!(self.eggcc_config.schedule, Schedule::Parallel, "Output egglog only works in parallel mode. Sequential mode does not use a single egraph");
-
                 let rvsdg = Optimizer::program_to_rvsdg(&self.prog_with_args.program)?;
                 let (dag, mut cache) = rvsdg.to_dag_encoding(true);
 
-                let schedule_steps = parallel_schedule();
-                if schedule_steps.len() != 1 {
-                    log::warn!("Parallel schedule had multiple steps! You may need to adjust the schedule to make eggcc tractable.");
-                }
+                // to deal with i64::MAX
+                let stop_after_n_passes = i64::min(
+                    self.eggcc_config.stop_after_n_passes,
+                    parallel_schedule().len() as i64,
+                );
+                let eggcc_config = EggccConfig {
+                    // stop before the last pass that user specified.
+                    stop_after_n_passes: stop_after_n_passes - 1,
+                    ..self.eggcc_config.clone()
+                };
+                let optimized = dag_in_context::optimize(&dag, &mut cache, &eggcc_config)
+                    .map_err(EggCCError::EggLog)?;
+
+                let schedules = parallel_schedule();
+                let last_schedule_step = schedules.last().unwrap();
```

Comment:
You're still using the wrong pass here- didn't you want the stop_after_n_passes pass?

Reply:
good catch, fixed!